### ISSUE TITLE
Feature: Control of function availability in Resource definition

### ISSUE THREAD

**Issue by dhnaranjo** [avo-hq/avo#1879]
### Feature
As an example, allow field definitions like:
```
field :things,
  as: :has_many,
  attach: false,
  detach: false
```
Or some other method of providing arguments directly on the field definition. I am not advocating specifically for that format for the arguments, just the ability to control that type of thing within the Resource file. Maybe something like 

```
field :things,
  as: :has_many,
  controls: [:attach, :detach, :destroy]
```

### Current workarounds
So many Policy classes.

### Additional context
What I am trying to do is configure a resource, not authorize a user to perform an action. There's only one class of user with access to the Avo-driven section of my site. Storing Avo resource configuration in an auth policy doesn't make any sense and is confusing.

**Comment by adrianthedev** [comment]
Yes. I hear you @dhnaranjo. We're a bit busy with launching the open beta. 
We'll pick this up a bit later.

One question that pops into mind is how would you authorize detaching and destroying at a row-level?

**Comment by dhnaranjo** [comment]
I think I would leave the existing authorization behavior in place. Auth-style answers the question "can this action be done in this specific case?", the new proposal would answer the question "is this a feature I am using?"

It would be in keeping with the general Avo config style to accept a proc for a more lightweight solution to context-specific behavior, like a:
```
field :things,
  as: :has_many,
  controls: {
    destroy: -> { !associated.immutable? }
  }
```

But I think it would simplify things if context-specific behavior was only available through the Auth system. Less ambiguity.

**Comment by adrianthedev** [comment]
I think this is fixed by authz and custom controls.
Will reopen when we have a specific case in mind.
